sphinx/util/inspect.py

```diff
@@ -116,6 +116,15 @@ def getargspec(func: Callable) -> Any:
                                kwonlyargs, kwdefaults, annotations)
 
 
+def unwrap(obj: Any) -> Any:
+    """Get an original object from wrapped object (wrapped functions)."""
+    try:
+        return inspect.unwrap(obj)
+    except ValueError:
+        # might be a mock object
+        return obj
```

Comment:
@tk0miya this looks good, but maybe log DEBUG/INFO?

```suggestion
        logger.debug('Giving up trying to unwrap %r', obj)
        return obj
```

Reply:
I doubt the log will be helpful. Do you want to see it?

Reply:
I'm fine not seeing it, I was just thinking it could be useful if you were already debugging.